src/config/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+set(SOURCES bootstrap.cc
+            config.cc)
+
+set(HEADERS bootstrap.h
```

Comment:
I am not sure about the config component since I think it should only be used by the primary aktualizr and not shared with the secondary executable which has its own config. It's almost possible now to link the secondary without config now (only need to move `PackageConfig` to `package_manager`).

config_utils is for everyone though (in utilities..?). 

Perhaps `Config` should be named `AktualizrConfig` now that we have other configs. That might be a big change though.

Reply:
Is the question whether we want the config object to be at the top or the bottom of the dependency graph?

If we put it at the top (therefore the implementation modules depend on it), then either it will need to contain both primary and secondary config, or we have to duplicate the implementation inheritance tree (config_common/config_primary/config_secondary).

The other option would be to put the config object at the bottom of the dependency graph (i.e. it depends on everything). This would let us put the configuration inside the modules that use it, but would tangle the configuration with the implementation.

The model I have in my head is that configuration goes through a set of transformations:

- sota.toml (byte string)
-  Boost property tree (untyped tree-ish thing)
- Typed configuration holder object (`Config`). This could/should be validated for consistency.
- Dependency injection then uses `Config` to instantiate the actual system.

Reply:
It used to follow the first model (everything depends on it) but I've moved it to the bottom (almost completely) when creating the secondary config since I felt it was a bit cleaner and make it easier to have self-contained components.

We can still go back though.

Reply:
Summary of a F2F discussion: We will put the config at the bottom of the dependency tree. This means that:

* I'll undo the change to move config into a module
* I'll move config_utils into utilities
* In future refactorings we will break up the config object into smaller pieces, and move these into the modules.